service-base/src/test/java/org/eclipse/hono/service/VertxBasedHealthCheckServerTest.java

```diff
@@ -99,7 +99,7 @@ void testHealthCheckServerWithInsecureEndpoint(final Vertx vertx, final VertxTes
             .map(ok -> getWebClient(vertx, server, false))
             .compose(httpClient -> checkHealth(ctx, httpClient, "/liveness"))
             .compose(httpClient -> checkHealth(ctx, httpClient, "/readiness"))
-            .onComplete(ctx.completing());
+            .onComplete(ctx.succeeding(v -> testsDone.flag()));
```

Comment:
why should this change be necessary?

Reply:
Because `registerHealthChecks(ctx, server, 1);` a few lines above creates checkpoints (`1` is the number of checkpoint passes).
I've now renamed the method to `registerHealthCheckCheckpoints` to make this more obvious.

Reply:
Ah, my bad, didn't see that ...